lib/ruby_lsp/executor.rb

```diff
@@ -0,0 +1,343 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  # This class dispatches a request execution to the right request class. No IO should happen anywhere here!
+  class Executor
```

Comment:
Should this be named `Server` instead?

Reply:
From my domain experience `Executor` makes sense for this class. With the functionality change of `Handler` I could see that being updated to `Server` as it feels like that's what it semantically represents now.

Reply:
Renamed it to `Server` 👍.